### PR TITLE
port serveral middlewares to konaf config

### PR DESCRIPTION
**Pull request by matoszz**
- converts several middleware to use serveropts over statically appended
- adds koanf tags to structs

### CODE REVIEW

File: pkg/middleware/cors/cors.go [hunk 1]
```diff
@@ -11,9 +11,11 @@ import (
 
 // Config holds the cors configuration settings
 type Config struct {
-	// Skipper defines a function to skip middleware.
-	Skipper  middleware.Skipper
-	Prefixes map[string][]string
+	Enabled        bool                `json:"enabled" koanf:"enabled" default:"true"`
```
Comment (golanglemonade) [inline pkg/middleware/cors/cors.go:14]: Add comments to these so the descriptions show up in the config docs

File: pkg/middleware/mime/mime.go [hunk 2]
```diff
@@ -10,13 +10,13 @@ import (
 	"github.com/datumforge/echox/middleware"
 )
 
-// Config defines the config for Mime middleware.
+// Config defines the config for Mime middleware
 type Config struct {
```
Comment (golanglemonade) [inline pkg/middleware/mime/mime.go:14]: this config is not included in `config/config.go`

File: pkg/middleware/redirect/redirect.go [hunk 3]
```diff
@@ -9,10 +9,11 @@ import (
 
 // Config contains the types used in executing redirects via the redirect middleware
 type Config struct {
-	// Skipper defines a function to skip middleware.
-	Skipper   middleware.Skipper
-	Redirects map[string]string
-	Code      int
+	Enabled bool `json:"enabled" koanf:"enabled" default:"true"`
```
Comment (golanglemonade) [inline pkg/middleware/redirect/redirect.go:12]: Add comments 🙏

File: pkg/middleware/secure/secure.go [hunk 4]
```diff
@@ -1,24 +1,55 @@
...
+	Enabled               bool               `json:"enabled" koanf:"enabled" default:"true"`
+	Skipper               middleware.Skipper `json:"-" koanf:"-"`
+	XSSProtection         string             `json:"xssprotection" koanf:"xssprotection" default:"1; mode=block"`
+	ContentTypeNosniff    string             `json:"contenttypenosniff" koanf:"contenttypenosniff" default:"nosniff"`
+	XFrameOptions         string             `json:"xframeoptions" koanf:"xframeoptions" default:"SAMEORIGIN"`
+	HSTSPreloadEnabled    bool               `json:"hstspreloadenabled" koanf:"hstspreloadenabled" default:"false"`
+	HSTSMaxAge            int                `json:"hstsmaxage" koanf:"hstsmaxage" default:"31536000"`
+	ContentSecurityPolicy string             `json:"contentsecuritypolicy" koanf:"contentsecuritypolicy" default:"default-src 'self'"`
+	ReferrerPolicy        string             `json:"referrerpolicy" koanf:"referrerpolicy" default:"same-origin"`
+	CSPReportOnly         bool               `json:"cspreportonly" koanf:"cspreportonly" default:"false"`
```
Comment (golanglemonade) [inline pkg/middleware/secure/secure.go:19]: Same, all these fields need to be commented

File: internal/httpserve/serveropts/option.go [hunk 5]
```diff
@@ -390,3 +379,40 @@ func WithRateLimiter() ServerOption {
...
+func WithCacheHeaders() ServerOption {
+	return newApplyFunc(func(s *ServerOptions) {
+		if s.Config.Settings.Server.CacheControl.Enabled {
+			cacheConfig := s.Config.Settings.Server.CacheControl
+			s.Config.DefaultMiddleware = append(s.Config.DefaultMiddleware, cachecontrol.NewWithConfig(cacheConfig))
+		}
+	})
+}
+
+func WithCORS() ServerOption {
```
Comment (golanglemonade) [inline internal/httpserve/serveropts/option.go:412]: missing comment

**Comment by matoszz** [comment]
addresses https://github.com/datumforge/datum/issues/643

**Review by golanglemonade — APPROVED** [review]

**Pull request merged**